src/stat.cpp

```diff
@@ -0,0 +1,32 @@
+#include <cmath>
+#include "utiltime.h"
+#include "stat.h"
+
+
+ExponentialMovingAverage::ExponentialMovingAverage(const double _interval,
+    const size_t _time_units) :
+    interval(_interval), time_units(_time_units) {}
```

Comment:
suggest throwing `std::invalid_argument` if `_interval == 0 || _time_units == 0`. You risk divide by zero in `::update` if not.